### ISSUE TITLE
"Object reference not set to an instance of an object" error

### ISSUE THREAD

**Issue by Bippers** [duplicati/duplicati#1933]
I have:
- [x] searched open and closed issues for duplicates

---
### Version info

**Duplicati Version:** 2.0.1.23_canary_2016-09-16
**Operating System:** Mac OS X 10.11.6 (El Capitan) / Windows 7
**Backend:**
### Bug description

Creating a backup from Mac OS X and then trying to restore it on Windows gives an error of "Object reference not set to an instance of an object". Also happens when doing it the other way around.
### Steps to reproduce
- Backup some data from Mac OS X to a cloud provider (I'm using Amazon Cloud Drive)
- Try to restore the backup from a Windows PC
- Restore fails and error is shown
- Happens when creating the backup from Windows and restoring from Mac

**Actual result:** Error above is shown
**Expected result:** File list is shown and files can be restored correctly (either selectively or all of them)
### debug log

```
Sep 18, 2016 3:18 PM: Request for http://127.0.0.1:8200/api/v1/backup/6a977102-49dd-4539-896a-735efdf56fe2/files/?prefix-only=false&folder-contents=true&time=2016-09-18T15:04:25%2B02:00&filter= gave error
System.NullReferenceException: Object reference not set to an instance of an object
  at Duplicati.Server.WebServer.RESTMethods.Backup.SearchFiles (IBackup backup, System.String filterstring, Duplicati.Server.WebServer.RESTMethods.RequestInfo info) <0xa1cd3a0 + 0x0009c> in <filename unknown>:0 
  at Duplicati.Server.WebServer.RESTMethods.Backup.GET (System.String key, Duplicati.Server.WebServer.RESTMethods.RequestInfo info) <0x98f5928 + 0x0032b> in <filename unknown>:0 
  at Duplicati.Server.WebServer.RESTHandler.DoProcess (Duplicati.Server.WebServer.RESTMethods.RequestInfo info, System.String method, System.String module, System.String key) <0x3c0b248 + 0x005d8> in <filename unknown>:0 

Sep 18, 2016 3:18 PM: Reporting error gave error
System.ObjectDisposedException: Cannot write to a closed TextWriter.
  at System.IO.StreamWriter.Flush (Boolean flushStream, Boolean flushEncoder) <0x34b9288 + 0x00043> in <filename unknown>:0 
  at System.IO.StreamWriter.Flush () <0x3717498 + 0x0002f> in <filename unknown>:0 
  at Duplicati.Server.WebServer.RESTHandler.DoProcess (Duplicati.Server.WebServer.RESTMethods.RequestInfo info, System.String method, System.String module, System.String key) <0x3c0b248 + 0x0145b> in <filename unknown>:0 
```
##

**Comment by Bippers** [comment]
Just a quick question. Do you know if this will get fixed? Currently it is the only thing stopping me from using Duplicati2 to backup my things. If I start the backup process now will a future version of Duplicati be able to restore it on both Windows & Mac? Thanks.

**Comment by jheyneman** [comment]
Just experienced this today in testing my backups. I did a backup from a Linux machine to Amazon Cloud Drive, and tried to restore using my Windows laptop. 

`Oct 17, 2016 3:12 PM: Request for http://127.0.0.1:8200/api/v1/backup/976bac24-db5e-4435-9824-6db65c99de95/files/?prefix-only=false&folder-contents=true&time=2016-09-26T19:39:43-04:00&filter= gave error
System.NullReferenceException: Object reference not set to an instance of an object.
   at Duplicati.Server.WebServer.RESTMethods.Backup.SearchFiles(IBackup backup, String filterstring, RequestInfo info)
   at Duplicati.Server.WebServer.RESTMethods.Backup.GET(String key, RequestInfo info)
   at Duplicati.Server.WebServer.RESTHandler.DoProcess(RequestInfo info, String method, String module, String key)

Oct 17, 2016 3:12 PM: Reporting error gave error
System.ObjectDisposedException: Cannot write to a closed TextWriter.
   at System.IO.__Error.WriterClosed()
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at Duplicati.Server.WebServer.RESTHandler.DoProcess(RequestInfo info, String method, String module, String key)`

**Comment by Jop-V** [comment]
Attempting to restore a Windows backup to a Linux machine also gives this error.

**Comment by Nerdy314159265** [comment]
I also can confirm this bug. I'm testing backing up Duplicati from a machine running FreeNAS (based on FreeBSD) to a machine running Windows and have the same error. The reverse also has the same issue.

**Comment by tischnu** [comment]
Bug still exists in 2.0.1.31_canary_2016-11-03.
**Workaround:** Enter something into the search box. I can then select and restore all results - it's not possible to tick boxes on specific results...

**Comment by vitekf** [comment]
I tried Tischnu`s trick, restoring looks sucessfull, but nothing was restored on Windows
(backuped from Synology to SFTP, restoring from Windows)

Synology restore (it works well):
![dup3](https://cloud.githubusercontent.com/assets/23286396/20623159/95cde812-b306-11e6-90db-2aef88ec8d1b.PNG)

Windows restore (after Error popup and put "v" into search box)
![dup4](https://cloud.githubusercontent.com/assets/23286396/20623166/9e037556-b306-11e6-94ca-94446af8d0c2.PNG)
I can select files, click Continue, pick new local location, click Restore... Duplicati finish with OK message, but no files were restored

**Comment by cyrilpawelko** [comment]
Same behavior with 2.0.1.30_experimental_2016-11-03 : Saved from linux (Asustor NAS) and trying to restore on Windows 10.

**Comment by edhemphill** [comment]
I'm getting the same behavior on the command line with a command like:
```
duplicati-cli \
  --aws_secret_access_key="+XXXX" \
  --aws_access_key_id="XXX" \
  --s3-ext-signatureversion=4 --passphrase=XXX --use-ssl --s3-location-constraint=us-east-1 \
  --prefix=%2fmnt%2fstorage1%2fdocker-data%2fdocker \
  backup s3://this-bucket/folder \
  docker
```

```
Backup started at 12/13/2016 4:06:00 PM
Checking remote backup ...
  Listing remote folder ...
Scanning local files ...
Fatal error => Object reference not set to an instance of an object
  105 files need to be examined (293.99 MB)
  0 files need to be examined (0 bytes)

Object reference not set to an instance of an object
```

**Comment by kenkendk** [comment]
This should be fixed with `2.0.1.39` and later.
Re-open if this is not the case.
